packages/product-catalogue/src/components/ProductForm/index.tsx

```diff
@@ -0,0 +1,284 @@
+import React, { useState } from 'react';
+import { Form, SubmitButton, FormItem, Radio, Input, InputNumber } from 'formik-antd';
+import { Formik } from 'formik';
+import { ProductCatalogue } from '../../ducks/productCatalogue';
+import { PlusOutlined } from '@ant-design/icons';
+import { Upload, Space, Button } from 'antd';
+import { UploadChangeParam, UploadFile } from 'antd/lib/upload/interface';
+import { postProduct, putProduct } from '../..//helpers/dataLoaders';
+import { sendErrorNotification, sendSuccessNotification } from '@opensrp/notifications';
+import * as Yup from 'yup';
+import { CATALOGUE_LIST_VIEW_URL } from '../../constants';
+import { Redirect, useHistory } from 'react-router';
+import { CommonProps, defaultCommonProps } from '../../helpers/common';
+
+/** type describing the fields in the product catalogue form */
+type ProductFormFields = Omit<ProductCatalogue, 'serverVersion' | 'productPhoto'> & {
+  productPhoto: string | File;
+};
+
+const defaultInitialValues: Partial<ProductFormFields> = {
```

Comment:
Does it need to be Partial if we are going to have properties set as undefined?

Reply:
will remove partial, remove the redundancy.